vortex-array/src/buffer.rs

```diff
@@ -0,0 +1,58 @@
+use std::ops::Deref;
+
+use vortex_buffer::Buffer;
+use vortex_error::{vortex_panic, VortexExpect};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub struct ArrayBuffer {
```

Comment:
Why not have this in vortex-buffer?

Reply:
Because I want to get rid of it! StringBuffer, ScalarBuffer, and ArrayBuffer can all live in vortex-array/src/buffer I think?